iotdb-core/ainode/iotdb/ainode/core/inference/inference_request_pool.py

```diff
@@ -107,42 +111,77 @@ def _requests_activate_loop(self):
             self._activate_requests()
 
     def _step(self):
-        requests = self._request_scheduler.schedule_step()
-        # TODO: We need a batcher to accelerate the concurrent inference
-        for request in requests:
+        all_requests: list[InferenceRequest] = self._request_scheduler.schedule_step()
+
+        grouped_requests = defaultdict(list)
+        for req in all_requests:
+            key = (req.inputs.shape[1], req.max_new_tokens)
+            grouped_requests[key].append(req)
+        grouped_requests = list(grouped_requests.values())
+
+        for requests in grouped_requests:
+            batch_inputs = self._batcher.batch_request(requests).to(self.device)
             if self.model_id == "sundial":
-                request.inputs = request.inputs.to(self.device)
-                output = self._model.generate(
-                    request.inputs,
-                    max_new_tokens=request.max_new_tokens,
+                batch_output = self.model.generate(
+                    batch_inputs,
+                    max_new_tokens=requests[0].max_new_tokens,
                     num_samples=10,
                     revin=True,
                 )
-                request.output_tensor = request.output_tensor.to(self.device)
-                request.write_step_output(output[0].mean(dim=0))
+
+                offset = 0
+                for request in requests:
+                    request.output_tensor = request.output_tensor.to(self.device)
+                    b = request.batch_size
+                    output_i = batch_output[offset : offset + b]
```

Comment:
Rename them for more direct expression.
```suggestion
                    cur_batch_size = request.batch_size
                    cur_output = batch_output[offset : offset + b]
```